theano/gof/fg.py

```diff
@@ -402,7 +402,7 @@ def find_path_to(output_var, input_var):
                             assert path is not None
                             tr = getattr(r.tag, 'trace', [])
                             detailed_err_msg = ""
-                            if len(tr) > 0:
+                            if type(tr) is list and len(tr) > 0:
```

Comment:
Why is it necessary to check the type of `tr`? Could it be done using `isinstance` instead?